### ISSUE TITLE
nixfmt always updates mtime

### ISSUE THREAD

**Issue by felschr** [numtide/treefmt-nix#99]
**Describe the bug**

Running nixfmt always sets the mtime to the current date even if the contents didn't change.

**To Reproduce**
```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    
    flake-parts.url = "github:hercules-ci/flake-parts";
    flake-parts.inputs.nixpkgs.follows = "nixpkgs";
    
    treefmt-nix.url = "github:numtide/treefmt-nix";
    treefmt-nix.inputs.nixpkgs.follows = "nixpkgs";
  };

  outputs = inputs@{ flake-parts, ... }:
    flake-parts.lib.mkFlake { inherit inputs; } {
      imports = [
        inputs.treefmt-nix.flakeModule
      ];
      systems = ["x86_64-linux"];
      perSystem = { pkgs, ... }: {
        treefmt = {
          programs.nixfmt.enable = true;
          programs.nixfmt.package = pkgs.nixfmt;
        };
      };
    };
}
```

Add some nix files to the workspace and observe the mtime each time `treefmt --no-cache` is run.

**Expected behavior**

treefmt-nix should not support nixfmt since it changes mtime and thus doesn't comply with idempotence requirement of the [formatter specification](https://numtide.github.io/treefmt/formatters-spec/).

Additionally it would be nice to have a list on https://numtide.github.io/treefmt/formatters/ with known unsupported formatters and their reasons for not being supported.

**System information**
```
Inputs:
├───flake-parts: github:hercules-ci/flake-parts/8e8d955c22df93dbe24f19ea04f47a74adbdc5ec
│   └───nixpkgs-lib follows input 'nixpkgs'
├───nixpkgs: github:NixOS/nixpkgs/6cee3b5893090b0f5f0a06b4cf42ca4e60e5d222
└───treefmt-nix: github:numtide/treefmt-nix/f4ce7d8660f57fe94092cf113c1f550acf16be53
    └───nixpkgs follows input 'nixpkgs'
```

**Comment by SebTM** [comment]
See referenced issue, I summarized the report and issue but I can't address it due to lack of Haskell knowledge myself.

**Comment by SebTM** [comment]
Resolved with v0.6.0 - but I can't close the issue :v:

**Comment by zimbatm** [comment]
nixos-unstable is still on nixfmt 0.5.0 at the moment

**Comment by SebTM** [comment]
NixOS 23.11 got released and includes nixfmt v0.6.0 by default ✌🏻
